pkg/registryclient/client.go

```diff
@@ -168,40 +164,28 @@ func WithTracing() Option {
 	}
 }
 
-// BuildCosignRemoteOption builds remote.Option for cosign client.
-func (c *client) BuildCosignRemoteOption(ctx context.Context) (remote.Option, error) {
-	gcrRemoteOpts, err := c.getGCRRemoteOption(ctx)
-	if err != nil {
-		return nil, err
-	}
-	gcrRemoteOpts = append(gcrRemoteOpts, gcrremote.WithUserAgent(userAgent))
-	return remote.WithRemoteOptions(gcrRemoteOpts...), nil
-}
-
-// BuildGCRRemoteOption builds []gcrremote.Option based on client.
-func (c *client) BuildGCRRemoteOption(ctx context.Context) ([]gcrremote.Option, error) {
-	return c.getGCRRemoteOption(ctx)
-}
-
-func (c *client) getGCRRemoteOption(ctx context.Context) ([]gcrremote.Option, error) {
-	remoteOpts := []gcrremote.Option{
+// Options returns remote.Option config parameters for the client
+func (c *client) Options(ctx context.Context) ([]gcrremote.Option, error) {
+	opts := []gcrremote.Option{
 		gcrremote.WithAuthFromKeychain(c.keychain),
 		gcrremote.WithTransport(c.transport),
 		gcrremote.WithContext(ctx),
+		gcrremote.WithUserAgent(userAgent),
```

Comment:
This is the user agent that is being passed here.
```golang
userAgent = fmt.Sprintf("cosign/%s (%s; %s)", version.GetVersionInfo().GitVersion, runtime.GOOS, runtime.GOARCH)
```

Since notary implementation also uses this function now, Is it okay to pass this to notary as well?

Reply:
Good point. I changed it to `Kyverno/%s (%s; %s) `.